src/app/inbound/mod.rs

```diff
@@ -54,9 +54,12 @@ where
         .layer(tls::client::layer(local_identity))
         .service(connect::svc());
 
+    let (_span_rx, trace_layer) = trace::layer(100);
```

Comment:
This looks generally good, but...

istm that you're going to want to pass the inbound/outbound server functions some interface that can receive traces, rather than returning an mpsc::Receiver from each.

As a very rough idea:

```rust
pub trait SpanSink {
  type Error: Into<Error>;

  fn poll_ready(&mut self) -> Poll<(), Self::Error>;

  fn send_span(&mut self, span: Span) -> Result<(), Self::Error>;
}
```